### PR TITLE
[ci] [ocaml] Bump edge compiler to Ocaml 4.13.0

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

**Comment by jfehrle** [comment]
IIUC the "edge compiler" is the maximum supported version of the compiler. If so, it would more understandable just to refer to the "maximum OCaml version".  The base version could be the "minimum OCaml version".

**Comment by ejgallego** [comment]
> IIUC the "edge compiler" is the maximum supported version of the compiler. If so, it would more understandable just to refer to the "maximum OCaml version". The base version could then be the "minimum OCaml version".

Good point, tho our testing strategy is not perfect, we usually test a compiler a lot, then we move to the next one in the edge branch. This has worked fine [few regressions on intermediate compilers] but it is far from perfect.

File: kernel/dune [hunk 1]
```diff
@@ -30,7 +30,7 @@
 ; In dev profile, we check the kernel against a more strict set of
 ; warnings.
 (env
- (dev (flags :standard -w +a-4-44-50)))
+ (dev (flags :standard -w +a-4-44-50-70)))
  ; (ocaml408 (flags :standard -w +a-3-4-44-50)))
```
Comment (SkySkimmer) [inline kernel/dune:34]: BTW is this comment still useful?

File: tools/configure/configure.ml [hunk 2]
```diff
@@ -133,7 +133,7 @@ let check_findlib_version prefs { CamlConf.findlib_version } =
     68: "This pattern depends on mutable state" no idea what it means, dune builds don't display it
 *)
 
-let coq_warnings = "-w +a-4-9-27-41-42-44-45-48-58-67-68"
+let coq_warnings = "-w +a-4-9-27-41-42-44-45-48-58-67-68-70"
```
Comment (SkySkimmer) [inline tools/configure/configure.ml:136]: The list above should be updated.

**Review by SkySkimmer — APPROVED** [review]

File: kernel/dune (hunk 1 above)
Comment (ejgallego) [reply kernel/dune]: Nope.

File: tools/configure/configure.ml (hunk 2 above)
Comment (ejgallego) [reply tools/configure/configure.ml:136]: Done.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**